models/model.js

```diff
@@ -33,18 +33,64 @@ exports.selectAllArticles = () =>{
 }
 
 
+     function selectArticleidComment (article_id){
+         if(article_id === NaN){
+            return Promise.reject({ status: 400, msg: 'Bad request'})
+         }
+      
+         return db.query('SELECT * FROM comments WHERE article_id = $1', [article_id])
+         .then(({rows})=>{
+            return rows
+         })
+      
+      
+      }
+     
 
- exports.selectArticleidComment = (article_id) => {
-   return db.query('SELECT * FROM articles WHERE article_id = $1', [article_id])
-     .then(({ rows }) => {
-       if (rows.length === 0) {
-         return Promise.reject({ status: 404, msg: 'Not Found' });
-       } if(!article_id){
-         return Promise.reject({ status: 404, msg: 'Not Found' });
-       }
-       return db.query('SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC', [article_id])
-         .then(({ rows }) => {
-           return rows;
-         });
-     });
- };
\ No newline at end of file
+      async function insertCommentToArticle(article_id, msg) {
+        let { username, body } = msg;
+        if (isNaN(article_id)) {
+          return Promise.reject({ status: 400, msg: "Invalid Id" });
+        }
+        if (!username) {
+          return Promise.reject({ status: 400, msg: "User not defined" });
+        }
+        username = username.toLowerCase();
+   
+        async function checkArticleIdAndUsernameExists(article_id, username) {
+          const queryArticle_id = await db.query(
+            "SELECT article_id FROM articles WHERE article_id = $1",
+            [article_id]
+          );
+          const queryUsername = await db.query(
+            "SELECT username FROM users WHERE username = $1",
+            [username]
+          );
+          if (queryArticle_id.rows.length === 0) {
+            return Promise.reject({ status: 404, msg: "Article not found" });
+          }
+          if (queryUsername.rows.length === 0) {
+            return Promise.reject({ status: 404, msg: "User not found" });
+          }
+        }
+      
+        if (!body.length) {
+          return Promise.reject({ status: 400, msg: "Invalid Request" });
+        }
+        const ifArticle_idAndUsernameExist = await checkArticleIdAndUsernameExists(
+          article_id,
+          username
+        );
+        if (ifArticle_idAndUsernameExist) {
+          return ifArticle_idAndUsernameExist;
+        }
+      
+        const result = await db.query(
+          "INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING body;",
+          [article_id, username, body]
+        );
+        const { rows } = result;
+        return rows[0];
+      }
+
```

Comment:
This is really extensive work Nyene! You've obviously put alot of time into this. However, PSQL throws it's own error if the username doesn't exist, same with the Id. You can handle these in a psql handler and get rid of the checkExists function. It's best to display an understanding of automated error handling to employers. 

Remove the checkExists and handle the errors with middleware instead :)